lib/sycamore/sycamore/data/table.py

```diff
@@ -143,6 +145,96 @@ def from_dict(cls, dict_obj: dict[str, Any]) -> "Table":
 
         return ret
 
+    # TODO: There are likely edge cases where this will break or lose information. Nested or non-contiguous
+    # headers are one likely source of issues. We also don't support missing closing tags (which are allowed in
+    # the spec) because html.parser doesn't handle them. If and when this becomes an issue, we can consider
+    # moving to the html5lib parser.
+    @classmethod
+    def from_html(cls, html_str: Optional[str] = None, html_tag: Optional[Tag] = None) -> "Table":
+        """
+        Constructs a Table object from a well-formated HTML table.
+
+        Args:
+          html_str: The html string to parse. Must be enclosed in <table></table> tags.
+          html_tag: A BeatifulSoup tag corresponding to the table. One of html_str or html_tag must be set.
+        """
+
+        # TODO: This doesn't account for rowgroup/colgroup handling, which can get quite tricky.
+
+        if (html_str is not None and html_tag is not None) or (html_str is None and html_tag is None):
+            raise ValueError("Exactly one of html_str and html_tag must be specified.")
+
+        if html_str is not None:
+            html_str = html_str.strip()
+            if not html_str.startswith("<table>") or not html_str.endswith("</table>"):
+                raise ValueError("html_str must be a valid html table enclosed in <table></table> tags.")
+
+            root = BeautifulSoup(html_str, "html.parser")
+        elif html_tag is not None:
+            if html_tag.name != "table":
+                raise ValueError(f"html_tag must correspond to a valid <table> tag. Got {html_tag.name}")
+            root = html_tag
+        else:
+            # Should be unreachable
+            raise RuntimeError(f"Unable to process from_html parameters: html_str={html_str} html_tag={html_tag}")
+
+        cur_col = 0
+        cur_row = -1
+
+        cells = []
+        caption = None
+
+        # Traverse the tree of elements in a pre-order traversal.
+        for tag in root.find_all(recursive=True):
+            if tag.name == "tr":
+                cur_row += 1  # TODO: Should this be based on rowspan?
+                cur_col = 0
+
+            elif tag.name in {"td", "th"}:
+                # We allow a missing tr for the first row to handle the case when
+                # they have a thead.
```

Comment:
is it possible to have multiple theads? Can a single table contain 2 header rows

Reply:
yep